### PR TITLE
[release]Develop

### PR DESCRIPTION
**Pull request by bwaj287**
finally

### CODE REVIEW

**Comment by bwaj287** [comment]
for release CR

File: linux_sql/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+## Architecture and Design
+1) Describtion of the nodes/server relationship:   
+![Diagram](https://user-images.githubusercontent.com/33403067/81481156-0fbc3480-91fc-11ea-96b8-25a345fe6676.png)
+2) Describtion of the tables   
+**host_info:** The script collects hardware specification data and then insert the data to the psql instance, collects the system info such as cpu_number, cou_architecture, cpu_model, cou_cache, cpu_speed, total_memory infomation  
+**host_usage:** The script collects server usage data and then insert the data into the psql database. The script will be executed every minute using Linux crontab, it collects the spu usage infomation such as cpu_idle, cou_kernel_usage, disk_io_usage,dick_available infomation.  
+3) Describtion of the scripts  
+**docker_psql.sh:** This script offers a automate start/stop/create operation for the psql container, it returns error message if the certain condition is not meet.  
+**host_info.sh:** This script grep the system infomation and put into the psql database.
+**host_usage.sh:** The script collects server usage data and then insert the data into the psql database.
```
Comment (jarviscanada) [inline linux_sql/README.md]: fix markdown..missing a newline..make sure you check the rendered markdown from github webpage
image

File: linux_sql/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+## Introduction
+The Jarvis Linux Cluster Administration (LCA) team manages a Linux cluster of 10 nodes/servers which are running CentOS 7. 
+The idea is  to monitor the system usage of the this nodes and server that helps the LCA team to meet their business needs. 
+I need to implement the program using bash scripts and the PostgreSQL database in order to monitor the system hardware infomation and cpu resource, and also write SQL queries to answer some business need.
```
Comment (jarviscanada) [inline linux_sql/README.md]: you implemented

File: linux_sql/scripts/.bashrc [hunk 3]
```diff
@@ -0,0 +1 @@
+export PGPASSWORD = password
```
Comment (jarviscanada) [inline linux_sql/scripts/.bashrc]: remove this. never commit password...you could get fired :(

File: linux_sql/scripts/average_memusage.sql [hunk 4]
```diff
@@ -0,0 +1,8 @@
+select t,idd,name,avg(usage)
+from (
```
Comment (jarviscanada) [inline linux_sql/scripts/average_memusage.sql]: put all your queries into `sql/queries.sql`

File: linux_sql/scripts/hardware_info.sql [hunk 5]
```diff
@@ -0,0 +1,4 @@
+SELECT cpu_number,id AS host_id,total_mem AS "total_mem(x100MB)"
```
Comment (jarviscanada) [inline linux_sql/scripts/hardware_info.sql]: sql/queries.sql

File: linux_sql/scripts/host_info.sh [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+cpu_model=$(echo "$lscpu_out"  | egrep "Model name:" | awk -F':' '{print $2}' | xargs)
+cpu_mhz=$(echo "$lscpu_out"  | egrep "MHz:" | awk '{print $3}' | xargs)
+l2_cache=$(echo "$lscpu_out"  | egrep "L2 cache:" | awk '{print $3}' | sed 's/[^0-9]*//g' | xargs)
+total_mem=$(echo "$mem_info"  | egrep "Mem:" | awk '{print $2}' | sed 's/[^0-9]*//g' | xargs)
+timestamp=$(date +%F_%T)
+
+#Executing SQL from shell scripts
+psql -h $psql_host -U $psql_user -d $db_name -p $psql_port << EOF
+INSERT INTO host_info (hostname, cpu_number, cpu_architecture, cpu_model, cpu_mhz, l2_cache, total_mem, timestamp) VALUES ('$hostname', '$cpu_number', '$cpu_architecture', '$cpu_model', '$cpu_mhz', '$l2_cache', '$total_mem', '$timestamp');
+EOF
```
Comment (jarviscanada) [inline linux_sql/scripts/host_info.sh:35]: exit $?

File: linux_sql/scripts/host_usage.sh [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+memory_free=$(echo "$mem_info"  | egrep "Mem:" | awk '{print $4}' | sed 's/[^0-9]\.*//g' |  xargs)
+cpu_idle=$(echo "$cpu_info"  | egrep "1" | awk '{print $15}' | xargs)
+cpu_kernel=$(echo "$cpu_info"  | egrep "1" | awk '{print $14}' | xargs)
+disk_io=$(echo "$disk_info"  | egrep "sda" | awk '{print $10}' | sed 's/[^0-9]\.*//g' | xargs)
+disk_available=$(df -h /dev/sda1 --output=source,fstype,size,used,avail,pcent | egrep "/dev/sda1" | awk '{print $5}' | sed 's/[^0-9]\.*//g' | xargs)
+
+#Executing SQL from shell scripts
+psql -h $psql_host -U $psql_user -d $db_name -p $psql_port  << EOF
+INSERT INTO host_usage (host_id, memory_free, cpu_idle, cpu_kernel, disk_io, disk_available) VALUES ((SELECT id FROM host_info), '$memory_free', '$cpu_idle', '$cpu_kernel', '$disk_io', '$disk_available');
+EOF
```
Comment (jarviscanada) [inline linux_sql/scripts/host_usage.sh:31]: exit $?

File: linux_sql/scripts/test.sh [hunk 8]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+lscpu_out=`lscpu`
+
```
Comment (jarviscanada) [inline linux_sql/scripts/test.sh:4]: remove

File: linux_sql/README.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+**host_info.sh:** This script grep the system infomation and put into the psql database.
+**host_usage.sh:** The script collects server usage data and then insert the data into the psql database.
+
+## Usage
+1) The `ddl.sql` file in SQL are used to create database and to define the type and structure of the data that will be stored in a database. by excuteing in the psql database `\i ddl.sql`, the two tables are automaticely generated.
+2) by running the `./host_info.sh -h "localhost" -U "postgres" -d "host_agent" -p 5432 `  the system infomation is collect and copied into query.  
+3) by running the `./host_usage.sh -h "localhost" -U "postgres" -d "host_agent" -p 5432 `  the system infomation is collect and copied into query.  
+4) using bash command `crontab -e` into crontable setup page and edit the crontable with command `* * * * * bash /home/centos/dev/jarvis_data_eng_shawn/linux_sql/scripts/host_usage.sh "localhost" 5432 "host_agent" "postgres"  >>/tmp/host_usage.log` the `host_usage.sh` script will collect the spu_usage every minute.
```
Comment (jarviscanada) [inline linux_sql/README.md]: README is too short..please see sample 

image

**Review by jarviscanada — APPROVED** [review]

**Review by jarviscanada — CHANGES_REQUESTED** [review]
fix all and request another review please

**Comment by bwaj287** [comment]
changed

**Pull request merged**